### PR TITLE
Update CONTRIBUTING.rst, fix tests and update Dockerfile. Related to #5096.

### PR DESCRIPTION
**Pull request by xirdneh**
The tests that were failing are fixed in this version as well as some updates on the `Dockerfile` and `docker-compose.yml` file Now tests can be run locally and travis-ci should be happy too (hopefully). 
`CONTRIBUTIN.rst` file has been updated to show in more detailed what and how to run to verify the project and PRs as per #5096 . 
A list of things to check before submitting a PR has also been added. We would need to do a release with the latest changes on `cyanide` for this to work, tho. 
I didn't want to pin a commit hash on the requirements and there's a PR that fixes a JSON serialization issue.
Without that fix the result tests will fail.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -175,7 +175,8 @@ and participate in the discussion.
 2) **Determine if your bug is really a bug**.
 
 You shouldn't file a bug if you're requesting support. For that you can use
-the :ref:`mailing-list`, or :ref:`irc-channel`.
+the :ref:`mailing-list`, or :ref:`irc-channel`. If you must submit a github
```
Comment (auvipy) [inline CONTRIBUTING.rst]: If you still need to open an issue

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -441,7 +438,7 @@ section in the GitHub guides.
 If you need to work on a different branch than the one git calls ``master``, you can
 fetch and checkout a remote branch like this::
 
-    git checkout --track -b 3.0-devel origin/3.0-devel
+    git checkout --track -b 5.0-devel origin/5.0-devel
```
[the remark this answers is not in the document]
Comment (auvipy) [reply CONTRIBUTING.rst]: new branch should generally against the upstream/master

[the remark this answers is not in the document]
Comment (xirdneh) [reply CONTRIBUTING.rst]: Should be upstream but this is just an example on how to checkout a different branch other than master.
A few paragraphs before this one shows how to setup a branch to upstream/master

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -483,17 +480,87 @@ Some useful commands to run:
 
 * ``make test``
 
-    To run the test suite
+    To run the test suite.
+    **Note:** This will run tests using python 2.7 by default.
```
[the remark this answers is not in the document]
Comment (xirdneh) [reply CONTRIBUTING.rst]: Good point, will do.

File: requirements/pkgutils.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-setuptools>=20.6.7
+setuptools==28.8.0
```
[the remark this answers is not in the document]
Comment (xirdneh) [reply requirements/pkgutils.txt]: No specific reason. This is the setup tools version we end up with when installing all the dependencies.
We probably need to upgrade all the dependencies' versions.
I wanted to do that on another PR but I can add it to this one if that's ok with everyone.

**Comment by thedrow** [comment]
Unit tests are failing...

**Comment by xirdneh** [comment]
@thedrow 
So, if somebody opens a github issue they should submit a PR with the test case, right?
Also, the unit tests are failing because now Travis is also installing `cyanide` and the JSONEnconder is broken there.
I submitted a PR that fixes that and it's being merged to master but we have to do a release since I didn't want to pin cyanide to a hash.
Integration tests are all working now :)

**Comment by auvipy** [comment]
I will start looking into updating cyanide tomorrow.

**Comment by xirdneh** [comment]
@auvipy Let me know if you need any help with that :)

**Comment by xirdneh** [comment]
@thedrow @auvipy I have another question about this.
Right now when using Travis-CI we are not using cyanide (correct me if I'm wrong, please).
I think it might be a good idea to start using cyanide to do some stress testing, thoughts?

**Comment by auvipy** [comment]
@thedrow could you push the release process here? If you can do so I will be able to help in releasing the other libraries like Django-celery-results, kombu etc

File: requirements/pkgutils.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (auvipy) [reply requirements/pkgutils.txt]: revert the change then

**Comment by thedrow** [comment]
Sure, will do.

**Comment by auvipy** [comment]
let's push this forward

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -225,6 +226,9 @@ spelling or other errors on the website/docs/code.
         confidential information like API tokens and authentication
         credentials.
 
+    E) If your issue is tagged with `Needs Verification` please provide a code sample
```
[the remark this answers is not in the document]
Comment (thedrow) [reply CONTRIBUTING.rst]: @xirdneh Ping?

**Comment by xirdneh** [comment]
@auvipy I'll push the last updates today.

**Comment by xirdneh** [comment]
The unit tests are failing because this changed `tox` to install `cyanide`, which is what the `Dockerfile` and that way we are running tests the same way in our CI as well as in the docker image we are offering.
Cyanide has a bug that it cannot serialize `datetime` objects. This is fixed in the latest master but there hasn't been a release.
We can remove the use of `cyanide` in the CI for the time being and remove it from the requirements and then add it once it's stable.
`cyanide` is used by kombu if it's installed.

**Comment by xirdneh** [comment]
@auvipy To be sure, you just update the version on `.bumpversion.cfg` and then run
```
$ bumpvesion
```
right?

**Comment by xirdneh** [comment]
Would be great to get this out with v4.3

**Comment by auvipy** [comment]
we can disable cyanide temporarily for now.

**Comment by xirdneh** [comment]
@thedrow @auvipy All updates should be there now.
There's some issues with the integration tests, but that was introduced by the latest master.
Every test should be working once that's fixed.
I also updates the sphinx extension.

File: CONTRIBUTING.rst [hunk 6]
```diff
@@ -225,6 +226,9 @@ spelling or other errors on the website/docs/code.
         confidential information like API tokens and authentication
         credentials.
 
+    E) If your issue is tagged with `Needs Test Case` please provide a code sample
+       so we can verify when the issue is resolved.
```
Comment (auvipy) [inline CONTRIBUTING.rst]: need tests to avoid regression

Comment (thedrow) [reply CONTRIBUTING.rst]: Test cases are not unit or functional tests.
Test cases are simply the most minimal programs possible that will reproduce a problem.
Most of the time we can create integration tests from test cases.

Comment (xirdneh) [reply CONTRIBUTING.rst]: Ah, I see.
I'll make this clear in the docs. Thanks for clarifying :)

**Comment by xirdneh** [comment]
@auvipy @thedrow I've updated the Test Case description, both in the PR section and in the Labels sections.
Hopefully this can be shipped with v4.3

**Pull request merged**